File: 문제풀이/chap4-1/prob4-1-3.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 입력
		Scanner sc = new Scanner(System.in);
		double X1 = sc.nextDouble();
		double Y1 = sc.nextDouble();
		double R1 = sc.nextDouble();
		double X2 = sc.nextDouble();
		double Y2 = sc.nextDouble();
		double R2 = sc.nextDouble();
		
		// 원 중심과의 거리 구하기
		double D = Math.sqrt((X1 - X2) * (X1 - X2) + (Y1 - Y2) * (Y1 - Y2));
		
		// 답 출력
		if (D < Math.abs(R1 - R2)) {
			System.out.println("1");
		}
		else if (D == Math.abs(R1 - R2)) {
			System.out.println("2");
		}
		else if (D < R1 + R2) {
			System.out.println("3");
		}
		else if (D == R1 + R2) {
			System.out.println("4");
		}
		else {
			System.out.println("5");
		}
	}
}

File: 문제풀이/chap4-1/prob4-1-3.c
#include <stdio.h>
#include <math.h>
 
double X1, Y1, R1;
double X2, Y2, R2;
 
int main() {
	// 입력
	scanf("%lf%lf%lf", &X1, &Y1, &R1);
	scanf("%lf%lf%lf", &X2, &Y2, &R2);
	
	// 원 중심과의 거리 구하기
	double d = sqrt((X1 - X2) * (X1 - X2) + (Y1 - Y2) * (Y1 - Y2));
	
	// 답 출력
	if (d < abs(R1 - R2)) printf("1\n");
	else if (d == abs(R1 - R2)) printf("2\n");
	else if (d < R1 + R2) printf("3\n");
	else if (d == R1 + R2) printf("4\n");
	else printf("5\n");
	return 0;
}